### PR TITLE
Features/add client sending reuqest

### PR DESCRIPTION
**Pull request by zhaoyuxuan**
Add part for client sending request to server to communicate and its test.
Update the `test_update_weights` , changing from passing None to empty numpy array as argument to `update_and_submit_weights` function

### CODE REVIEW

File: mozfldp/model.py [hunk 1]
```diff
@@ -95,6 +95,8 @@ def minibatch_update(self, X, y):
             init_intercept.shape, dtype=np.float64, order="C"
         )
 
+        self.set_training_classes(np.array(labels))
```
Comment (dzeber) [inline mozfldp/model.py]: This should actually be called once, externally by the user of the model object, rather than in the update function. Sorry, I realize this wasn't clearly documented. The flow I had in mind was:

- create a `SGDModel`
- call `set_training_classes` on the `labels` from the full dataset
- use the model in a simulation

In fact, the point of the SGDClassifier requiring the labels to be passed at some point is in case the dataset used for the `partial_fit` doesn't contain all the possible classes, which could very well be the case in a minibatch, so we would want to call it externally either way.

In other words, you can just revert this back and assume it has been handled externally without further changes :).

File: mozfldp/client.py [hunk 2]
```diff
@@ -69,8 +71,20 @@ def update_and_submit_weights(
...
+        client_data = {
+            "coefs": current_coef.tolist(),
+            "intercept": current_intercept.tolist(),
+            "num_samples": self._n
+        }
+
+        payload = json.dumps(client_data)
+
+        # send the post request to update the weights
+        api_endpoint = "http://0.0.0.0:8000/api/v1/ingest_client_data/{}".format(self._id)
```
Comment (dzeber) [inline mozfldp/client.py]: @crankycoder The URL probably shouldn't be hardcoded, but what do you think is the best way to handle it - pass in on instantiation?

**Review by dzeber — CHANGES_REQUESTED** [review]
Looks good aside from 2 points commented inline.

File: mozfldp/client.py (hunk 2 above)
Comment (crankycoder) [reply mozfldp/client.py]: Let's use https://pypi.org/project/python-decouple/ to pull hostname and port from enviroment variables and use defaults of `127.0.0.1` for host and `8000` for port.

The code will end up looking something like:
```
from decouple import config

HOSTNAME = config('FLDP_HOST', '127.0.0.1')
PORT = config('FLDP_PORT', 8000)

...

api_endpoint = "http://{hostname:s}:{port:d}/api/v1/ingest_client_data/{id:d}".format(hostname=HOSTNAME, port=PORT, id=self._id) 

```

**Comment by dzeber** [comment]
Made a few additional fixes and rewrote the main test case to use actual model updates.
